scripts/legacysurvey/build_parent_sample.py

```diff
@@ -139,25 +292,59 @@ def _processing_fn(args):
                 image.append(Cutout2D(images[band].data, position, size, wcs=wcs).data)
             image = np.stack(image, axis=0)
 
+            image_dim = image.shape[1:]
+            # Skip cutouts who don't have the expected dimension
+            if image_dim != size:
+                continue
+
             # Build inverse variance
             invvar = []
             for band in ['invvar-g', 'invvar-r', 'invvar-i', 'invvar-z']:
                 invvar.append(Cutout2D(images[band].data, position, size, wcs=wcs).data)
             invvar = np.stack(invvar, axis=0)
 
+            # Build cutout catalog and mask
+            cutout = Cutout2D(images["image-i"].data, position, size, wcs=wcs)
+            catalog_selector = CatalogSelector(group, cutout)
+            cutout_mask = catalog_selector.get_object_mask()
+            cutout_catalog = catalog_selector.get_brightest_object_catalog()
+
+            # Build model image
+            model_image_cutout = np.stack(
+                [
+                    Cutout2D(channel, position, size, wcs=wcs).data
+                    for channel in model_image
+                ]
+            )
+            model_image_cutout = np.moveaxis(model_image_cutout, 0, -1)
+
             # Build mask
             mask = Cutout2D(images['maskbits'].data, position, size, wcs=wcs).data
 
-            out_images.append({
-                    'object_id': np.array(f'{obj["BRICKNAME"]}-{obj["OBJID"]}', dtype=_utf8_filter_typeb),
-                    'gid': obj['gid'],
-                    'image_band': np.array([f.lower().encode("utf-8") for f in _filters], dtype=_utf8_filter_type),
-                    'image_ivar': invvar,
-                    'image_array': image,
-                    'image_mask': mask.astype('bool'),
-                    'image_psf_fwhm': np.array([obj[f'PSFSIZE_{b}'] for b in ['G', 'R', 'I', 'Z']]),
-                    'image_scale': np.array([_pixel_scale for f in _filters]).astype(np.float32),
-            })
+            obj_data = {
+                "object_id": np.array(
+                    f'{obj["BRICKNAME"]}-{obj["OBJID"]}', dtype=_utf8_filter_typeb
+                ),
+                "gid": obj["gid"],
+                "image_band": np.array(
+                    [f.lower().encode("utf-8") for f in _filters],
+                    dtype=_utf8_filter_type,
+                ),
+                "image_ivar": invvar,
+                "image_array": image,
+                "bit_mask": mask.astype("bool"),
```

Comment:
it would be better to keep the previous convention of `image_mask`, that's the convention we use in the other image datasets to refer to the "mask" field of an image

Reply:
because the convention for underscore is to mean a subfield of the input field.

Reply:
I went ahead and renamed the field, also renamed the field in all files on ceph, no need to rerun the data preparation script.